app/state_machines/configurable_state_machine/machine.rb

```diff
@@ -120,7 +121,7 @@ def trigger_event(event:, params:)
           CaseTransition.unset_most_recent(@kase)
           write_transition(event: event, to_state: to_state, to_workflow: to_workflow, params: params)
           @kase.update!(current_state: to_state, workflow: to_workflow)
-          execute_after_transition_method(event_config: event_config, user: params[:acting_user])
+          execute_after_transition_method(event_config: event_config, user: user)
```

Comment:
Is this really all that's needed? Or did another branch commit a fix?

Reply:
Should be just this.

When I ran through it the user being passed to the Predicate was nil which is why it was sending the emails (We check if the user != kase.responder_assignment, and nil isn't the responder so it always returned true and sent the email

Reply:
👍

Reply:
Actually, hold on, can't we have a quick test added for this?